### PR TITLE
simplify attribute implements (sync #5040)

### PR DESCRIPTION
**Pull request by jareguo**

### CODE REVIEW

File: cocos/core/utils/misc.ts [hunk 1]
```diff
@@ -176,12 +176,6 @@ export function isPlainEmptyObj_DEV (obj) {
     return isEmptyObject(obj);
 }
 
-export function cloneable_DEV (obj) {
```
Comment (jareguo) [inline cocos/core/utils/misc.ts:179]: ES5 class only, remove it

File: cocos/core/data/utils/attribute.ts [hunk 2]
```diff
@@ -253,24 +196,8 @@ export const CCString = new PrimitiveType('String', '');
...
-Callbacks: {
-    _onAfterProp: function (constructor, propName) {},
-    _onAfterGetter: function (constructor, propName) {}, (NYI)
-    _onAfterSetter: function (constructor, propName) {}, (NYI)
-}
- */
-
 // Ensures the type matches its default value
-export function getTypeChecker (type: string, attributeName: string) {
+export function getTypeChecker_ET (type: string, attributeName: string) {
```
Comment (jareguo) [inline cocos/core/data/utils/attribute.ts:200]: ET means this check function should be only used in editor or test environment

File: cocos/core/data/class.ts [hunk 3]
```diff
@@ -548,51 +527,37 @@ function parseAttributes (constructor: Function, attributes: IAcceptableAttribut
     if (type) {
         const primitiveType = PrimitiveTypes[type];
         if (primitiveType) {
-            result.push({
```
Comment (jareguo) [inline cocos/core/data/class.ts:551]: Remove all temp objects when parsing CCClass

File: cocos/core/data/class.ts [hunk 4]
```diff
@@ -516,29 +497,27 @@ const PrimitiveTypes = {
...
-    type: string;
-    _onAfterProp?: OnAfterProp;
+interface IParsedAttribute extends IAcceptableAttributes {
     ctor?: Function;
     enumList?: readonly any[];
     bitmaskList?: any[];
 }
-const tmpAttrs = [];
+type OnAfterProp = (constructor: Function, mainPropertyName: string) => void;
+const onAfterProps_ET: OnAfterProp[] = [];
```
Comment (jareguo) [inline cocos/core/data/class.ts:506]: ET means these function should be only used in editor or test environment

File: cocos/core/utils/js-typed.ts [hunk 5]
```diff
@@ -152,8 +152,8 @@ export function createMap (forceDictMode?: boolean): any {
     if (forceDictMode) {
         const INVALID_IDENTIFIER_1 = '.';
         const INVALID_IDENTIFIER_2 = '/';
-        map[INVALID_IDENTIFIER_1] = true;
-        map[INVALID_IDENTIFIER_2] = true;
+        map[INVALID_IDENTIFIER_1] = 1;
+        map[INVALID_IDENTIFIER_2] = 1;
```
Comment (shrinktofit) [inline cocos/core/utils/js-typed.ts]: What's the purpose?

Comment (jareguo) [reply cocos/core/utils/js-typed.ts]: 随手改的…… 减少包体……

Comment (shrinktofit) [reply cocos/core/utils/js-typed.ts]: 不需要吧，有uglify……不然你加个注释说1是干嘛吧

Comment (jareguo) [reply cocos/core/utils/js-typed.ts]: updated

**Pull request merged**